kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/vault/VaultEdek.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.resources.vault;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * A HashiCorp Vault Encrypted Dek.
+ * <br/>
+ * Note: HashiCorp Vault restricts the formation of the key name. For the rules, see the
+ * <a href="https://github.com/hashicorp/vault/blob/cd8cc4ed967703b89aa0721bca4fe1c43b214b00/sdk/framework/path.go#L19">GenericNameRegexp</a>
+ * function in the Vault implementation.
+ *
+ * @param kekRef - kek reference. In this implementation it is the key's name.
+ * @param edek - edek bytes
+ */
+record VaultEdek(String kekRef, byte[] edek) {
```

Comment:
Why do we need this duplicated in the system tests? If its just a question of visibility from `io.kroxylicious.kms.provider.hashicorp.vault` we can make that public.

Reply:
Sorry I see you mentioning this in your PR description.

Reply:
That's right. I needed to copy them because they are not public. If they become public in the next release, I can update the code to remove them and we will reduce the duplicity.